### ISSUE TITLE
Replace `learningRate` with `FixedRateScheduler` in `IUpdater` implementations

### ISSUE THREAD

**Issue by holgerbrandl** [deeplearning4j/deeplearning4j#4961]
The current implementations of `IUpdater` have a `learningRate` and a `rateSchduler`. However, providing a fixed rate and a scheduler seems redundant, since the latter is preferred.

1. A FixedRateScheduler should be used in case of a constant learning rate
2. The `learningRate` property should be removed from the IUpdater implementations where possible?

Alex Black said:
> a design like that would be a major headache for maintaining backward compatibility with serialized models from previous versions

To phase out the `learningRate` field while maintaining backwards compatibility, the corresponding public constructors and the field could be removed. Backward compatibility could be done with smthg like

```java
  private Sgd(@JsonProperty("learningRate") double learningRate,
                @JsonProperty("learningRateSchedule") ISchedule learningRateSchedule){

        if(learningRateSchedule == null){
            this.learningRateSchedule = new FixedRateScheduler(learningRate);
        }else {
            this.learningRateSchedule = learningRateSchedule;
        }
}
```
    
and so on for all the other IUpdaters.

Since the constructor argument `learningRate` is still present in the suggested constructor snippet from above, jackson would call it with the rate and a null schedule old serialized models using __just__ a fixed learning rate , and the logic from above would change the rate into a FixedRateScheduler. So the internal field could be imho removed entirely, but the code would ensure backward compatibility of the json format

To test it use
https://github.com/deeplearning4j/deeplearning4j/tree/46065ee814e5d9f6e6a86178db69d3d58cf4c661/deeplearning4j-core/src/test/java/org/deeplearning4j/regressiontest

**Comment by AlexDBlack** [comment]
@holgerbrandl were you planning on looking at this?
Not a problem if not - I just don't want to duplicate work if you are :)

**Comment by holgerbrandl** [comment]
Yes, but I still need to understand your PR process better by finishing my first PR https://github.com/deeplearning4j/deeplearning4j/pull/4975/ and by getting a better handle on the tests which I am not yet able to run properly (see https://github.com/deeplearning4j/deeplearning4j/issues/4978).

Once this is done, I'd be happy to proceed with this ticket.

**Comment by agibsonccc** [comment]
Closing — this issue is over 7 years old with no recent activity. The codebase has undergone a major rewrite. If this is still relevant, please open a new issue with current reproduction steps.
